src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -35,111 +33,119 @@ public static string InvokeJS(string str)
             return Interop.Runtime.CompileFunction(snippet);
         }
 
-        public static int New<T>(params object[] parms)
+        public static object GetGlobalObject(string? str = null)
         {
-            return Interop.Runtime.New(typeof(T).Name, parms);
+            return Interop.Runtime.GetGlobalObject(str);
         }
 
-        public static int New(string hostClassName, params object[] parms)
+        public static void DumpAotProfileData(ref byte buf, int len, string extraArg)
         {
-            return Interop.Runtime.New(hostClassName, parms);
+            Interop.Runtime.DumpAotProfileData(ref buf, len, extraArg);
         }
 
-        public static object GetGlobalObject(string? str = null)
+        public static int GetCsOwnedObjectJsHandle(JSObject target, int shouldAddInflight)
         {
-            return Interop.Runtime.GetGlobalObject(str);
+            if (shouldAddInflight != 0)
+            {
+                target.AddInFlight();
+            }
+            return target.JSHandle;
         }
 
-        public static void DumpAotProfileData (ref byte buf, int len, string extraArg)
+        internal static bool ReleaseCsOwnedObject(JSObject objToRelease)
         {
-            Interop.Runtime.DumpAotProfileData(ref buf, len, extraArg);
+            lock (_csOwnedObjects)
+            {
+                _csOwnedObjects.Remove(objToRelease.JSHandle);
+                Interop.Runtime.ReleaseCsOwnedObject(objToRelease.JSHandle);
+            }
+            return true;
         }
 
-        public static int BindJSObject(int jsHandle, bool ownsHandle, int mappedType)
+        internal static IntPtr CreateCsOwnedObject(JSObject proxy, string typeName, params object[] parms)
+        {
+            object res = Interop.Runtime.CreateCsOwnedObject(typeName, parms, out int exception);
+            if (exception != 0)
+                throw new JSException((string)res);
+
+            var jsHandle = (int)res;
+
+            lock (_csOwnedObjects)
+            {
+                _csOwnedObjects[jsHandle] = new WeakReference<JSObject>(proxy, trackResurrection: true);
+            }
+
+            return (IntPtr)jsHandle;
+        }
+
+        public static JSObject CreateCSOwnedObject(int jsHandle, int mappedType)
```

Comment:
These names differ slightly in case

Reply:
Thanks, renamed to `CreateCsOwnedProxy` the name clash is not helping here